src/cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -239,14 +239,14 @@ int VpiSignalObjHdl::initialise(const std::string &name,
     } else {
         m_num_elems = vpi_get(vpiSize, GpiObjHdl::get_handle<vpiHandle>());
 
-        if (GpiObjHdl::get_type() == GPI_STRING) {
+        if (GpiObjHdl::get_type() == GPI_STRING || type == vpiConstant ||
```

Comment:
It would save us a function call if we just compare type, vpiStringVar was the only thing that mapped here other than vpiStringConst, which is covered by the other checks

Reply:
Abstraction good. It might not be the only thing one day and the compiler should be doing its job and inlining that.